tests/pos/capturing.scala

```diff
@@ -0,0 +1,4 @@
+class MT(sf: MT => String) {
+  // `s` is retained as a field, but `sf` should not be.
```

Comment:
To actually test this, I suggest making a run test and using Java reflection like https://github.com/lampepfl/dotty/blob/master/tests/run/paramForwarding.scala#L46